tests/testsuite/freshness.rs

```diff
@@ -2381,8 +2385,21 @@ fn linking_interrupted() {
 
     // Interrupt the child.
     child.kill().unwrap();
-    // Note: rustc and the linker are still running, let them exit here.
-    conn.write(b"X").unwrap();
+    child.wait().unwrap();
+    // Note: rustc and the linker may still be running because we didn't kill
+    // the entire process group. Normally, when a user hits Ctrl-C, everything
+    // is killed. However, setting up process groups in a cross-platform test
+    // is a pain, and there's no easy way to know when everything has been
+    // killed. This write will tell them to exit, pretending that they died
+    // before finishing. Ignore the error, because (sometimes?) on Windows
+    // everything is already killed.
+    let _ = conn.write(b"X");
+    // Sleep a bit to allow rustc to clean up and exit. I have seen some race
+    // conditions on macOS where clang dies with `no such
+    // file...t1-HASH.rcgu.o`. I think what is happening is that the old rustc
+    // process is still running, and deletes the `*.o` files while the command
+    // below is trying to write them. Not sure if that is macOS-specific.
+    std::thread::sleep(std::time::Duration::new(2, 0));
```

Comment:
I think what you're mentioning above sounds right, since `child.kill()` only kills Cargo itself we've still got rustc running, and rustc, realizing the linker failed, will eventually clean up after itself. If that races with the next rustc clobbering the same files that can cause issues.

I think though it'd be best (if not too hard) to use process groups here to ensure that we kill the entire process tree, including rustc.